src/providers/cryptoauthlib/mod.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+//! Microchip CryptoAuthentication Library provider
+//!
+//! This provider is a hardware based implementation of PSA Crypto, Mbed Crypto.
+use super::Provide;
+use crate::key_info_managers::ManageKeyInfo;
+use derivative::Derivative;
+use log::trace;
+use std::collections::HashSet;
+use std::io::{Error, ErrorKind};
+use std::sync::{Arc, RwLock};
+use uuid::Uuid;
+
+use parsec_interface::operations::list_providers::ProviderInfo;
+use parsec_interface::requests::{Opcode, ProviderID, ResponseStatus, Result};
+
+const SUPPORTED_OPCODES: [Opcode; 0] = [];
+
+/// CryptoAuthLib provider structure
+#[derive(Derivative)]
+#[derivative(Debug, Clone, Copy)]
+pub struct Provider {
+    // device: rust_cryptoauthlib::AtcaDevice,
+}
+
+impl Provider {
+    /// Creates and initialise a new instance of CryptoAuthLibProvider
+    fn new(_key_info_store: Arc<RwLock<dyn ManageKeyInfo + Send + Sync>>) -> Option<Provider> {
+        Some(Provider {})
+    }
+}
+
+impl Provide for Provider {
+    fn describe(&self) -> Result<(ProviderInfo, HashSet<Opcode>)> {
+        trace!("describe ingress");
+        Ok((ProviderInfo {
+            // Assigned UUID for this provider: b8ba81e2-e9f7-4bdd-b096-a29d0019960c
+            uuid: Uuid::parse_str("b8ba81e2-e9f7-4bdd-b096-a29d0019960c").or(Err(ResponseStatus::InvalidEncoding))?,
+            description: String::from("User space hardware provider, utilizing MicrochipTech CryptoAuthentication Library for ATECCx08 chips"),
+            vendor: String::from("Arm"),
```

Comment:
Maybe the vendor could be `"Microchip Technology "`?

Reply:
As already mentioned on slack - this is a political decision at the moment it is safe to use "Arm" as a vendor.

Reply:
Ok agree!